pallets/loans-ref/src/lib.rs

```diff
@@ -771,56 +761,63 @@ pub mod pallet {
 			policy::find_rule(rules, |trigger| loan.check_write_off_trigger(trigger))
 		}
 
-		fn update_portfolio_valuation_with_pv(
+		fn find_mut_active_loan(
 			pool_id: PoolIdOf<T>,
-			portfolio: &mut types::PortfolioValuation<T::Balance>,
-			old_pv: T::Balance,
-			new_pv: T::Balance,
-		) -> DispatchResult {
-			let prev_value = portfolio.value();
-
-			portfolio.update_with_pv_diff(old_pv, new_pv)?;
-
-			if prev_value != portfolio.value() {
-				Self::deposit_event(Event::<T>::PortfolioValuationUpdated {
-					pool_id,
-					valuation: portfolio.value(),
-					update_type: PortfolioValuationUpdateType::Inexact,
-				});
-			}
-
-			Ok(())
+			loan_id: T::LoanId,
+			active_loans: &mut BoundedVec<ActiveLoan<T>, T::MaxActiveLoansPerPool>,
+		) -> Result<&mut ActiveLoan<T>, DispatchError> {
+			Ok(active_loans
+				.iter_mut()
+				.find(|loan| loan.loan_id() == loan_id)
+				.ok_or_else(|| {
+					if CreatedLoan::<T>::contains_key(pool_id, loan_id) {
```

Comment:
I think it will not be a worst-case because if you do not find it, you won't write in the ActiveLoans, so failing this extrinsic is cheaper than running it successfully. Anyways, the error makes sense to simplify things 👍🏻

Reply:
Done in [fa0b176](https://github.com/centrifuge/centrifuge-chain/pull/1311/commits/fa0b176f1db24d739f2e04c6abdf0aca8d9cc274)